File: 0539-minimum-time-difference/0539-minimum-time-difference.py
class Solution:
    def findMinDifference(self, timepoints: List[str]) -> int:
        res = []

        for time in timepoints:
            hour = time[:2]
            minutes = time[3:]

            res.append(int(hour)*60 + int(minutes))

        res.sort()
        ans = 24 * 60

        for i in range(len(res) -1):
            ans = min(ans, res[i+1]-res[i])

        return min(ans, 24*60-res[len(res)-1]+res[0])

File: 0539-minimum-time-difference/0539-minimum-time-difference.cpp
class Solution {
public:
    int findMinDifference(vector<string>& time) {
        vector<int> res(time.size());

        for(int i = 0; i<time.size(); i++)
        {
            string curr = time[i];

            int hour = stoi(curr.substr(0,2));
            int mins = stoi(curr.substr(3));

            res[i] = (hour*60)+mins;
        }

        sort(res.begin(), res.end());

        int ans = 60 * 24;

        for(int i = 0; i<res.size()-1; i++)
            ans = min(ans, res[i+1]-res[i]);

        return min(ans, 24*60-res[res.size()-1]+res[0]);
    }
};